File: Data/StateMachine/LoopStruct.cs
namespace mcsim.Data.StateMachine
{
    public struct LoopStruct
    {
        public string Initial { get; }
        public string Condition { get; }
        public string Update { get; }
        public string ConditionCVar { get; set; }
        public string UpdateCVar { get; set; }
        public string LoopVar { get; set; }

        public LoopStruct(string initial, string condition, string update, string conditionCVar, string updateCVar, string loopVar)
        {
            this.Initial = initial;
            this.Condition = condition;
            this.Update = update;
            this.ConditionCVar = conditionCVar;
            this.UpdateCVar = updateCVar;
            this.LoopVar = loopVar;
        }
    }
}

File: mcsim-js/src/loopStruct.mjs
/* eslint-disable @typescript-eslint/camelcase */
export class LoopStruct {
    constructor(initial, condition, update, condition_cvar, update_cvar, loopvar) {
        this._initial = initial;
        this._condition = condition;
        this._update = update;
        this._condition_cvar = condition_cvar;
        this._update_cvar = update_cvar;
        this._loopvar = loopvar;
    }

    get initial() {
        return this._initial;
    }

    set initial(value) {
        this._initial = value;
    }

    get condition() {
        return this._condition;
    }

    set condition(value) {
        this._condition = value;
    }

    get update() {
        return this._update;
    }

    set update(value) {
        this._update = value;
    }

    get condition_cvar() {
        return this._condition_cvar;
    }

    set condition_cvar(value) {
        this._condition_cvar = value;
    }

    get update_cvar() {
        return this._update_cvar;
    }

    set update_cvar(value) {
        this._update_cvar = value;
    }

    get loopvar() {
        return this._loopvar;
    }

    set loopvar(value) {
        this._loopvar = value;
    }
}
